t/t-config.sh

```diff
@@ -42,6 +42,81 @@ begin_test "default config"
 )
 end_test
 
+begin_test "config reads from repository"
+(
+  set -e
+  reponame="repository-config"
+  setup_remote_repo "$reponame"
+  mkdir $reponame
+  cd $reponame
+  git init
+  git remote add origin "$GITSERVER/$reponame"
+  git lfs env | tee env.log
+  grep "Endpoint=$GITSERVER/$reponame.git/info/lfs (auth=none)" env.log
+
+  git config --file=.lfsconfig lfs.url http://lfsconfig-file
+  git config --file=.lfsconfig lfs.http://lfsconfig-file.access lfsconfig
+  git add .lfsconfig
+  git commit -m 'Add file'
+  git push origin HEAD
+
+  git checkout -b side
+  git config --file=.lfsconfig lfs.url http://lfsconfig-file-side
+  git config --file=.lfsconfig lfs.http://lfsconfig-file-side.access lfsconfig
+  git add .lfsconfig
+  git commit -m 'Add file for side'
+  git push origin HEAD
+
+  mkdir "../$reponame-2"
+  cd "../$reponame-2"
+  git init
+  git remote add origin "$GITSERVER/$reponame"
+
+  git lfs env | tee env.log
+  grep "Endpoint=$GITSERVER/$reponame.git/info/lfs (auth=none)" env.log
+
+  git fetch origin
+  git symbolic-ref HEAD refs/remotes/origin/side
+  git show "HEAD:.lfsconfig"
+  git lfs env | tee env.log
+  grep "Endpoint=http://lfsconfig-file-side (auth=lfsconfig)" env.log
+
+  git read-tree refs/remotes/origin/main
+  git lfs env | tee env.log
+  grep "Endpoint=http://lfsconfig-file (auth=lfsconfig)" env.log
+)
+end_test
+
+begin_test "can read LFS file with name before .lfsconfig"
+(
+  set -e
+  reponame="early-file-config"
+  setup_remote_repo "$reponame"
+  mkdir $reponame
+  cd $reponame
+  git init
+  git remote add origin "$GITSERVER/$reponame"
+
+  git lfs track "*.bin"
+  git config --file=.lfsconfig lfs.url "$GITSERVER/$reponame.git/info/lfs"
+
+  echo "abc" > .bin
+  echo "def" > a.bin
+
+  git add .
+  git ls-files
```

Comment:
Just for my own understanding, how does the `git ls-files` work in this test?  Would it fail with an exit code under some circumstances if the new functionality wasn't working as expected?

Reply:
I believe that's a leftover debugging statement, so I'll remove it.